src/app/data/services/local_storage_service.spec.ts

```diff
@@ -0,0 +1,45 @@
+///<reference path="../../../../typings/browser/ambient/jasmine/index.d.ts"/>
+///<reference path="../../mocks/ajax_service_mock_data.ts"/>
+///<reference path="local_storage_service.ts"/>
+///<reference path="../models/content_block.ts"/>
+
+
+describe('LocalStorageService', function () {
+  const service = new LocalStorageService();
+  const hash:AjaxContentBlocksHash =  { blub: <any> { blub: '1' } };
+  const hash2:AjaxContentBlocksHash = { blub: <any> { blub2: '2' } };
+
+  describe('The content block saver', function () {
+    beforeEach(function() {
+      localStorage.clear();
+    });
+
+    it('can save a hash and it gets only updated if it is expired', function () {
+      const key = '1';
+      service.put(key, hash);
+
+      expect(localStorage.getItem(key)).toEqual(JSON.stringify(hash));
+
+      service.put(key, hash2);
+      expect(localStorage.getItem(key)).toEqual(JSON.stringify(hash));
+    });
+
+    it('can save a hash and it gets updated if it is expired', function () {
+      const key = '2';
+      service.put(key, hash);
+
+      expect(localStorage.getItem(key)).toEqual(JSON.stringify(hash));
+
+      spyOn(service, 'isValid').and.returnValue(false);
```

Comment:
couldn't we implement a method which returns Date.now() which we can stub? isValid isn't tested otherwise